src/Datasource/RulesChecker.php

```diff
@@ -307,7 +307,7 @@ protected function _checkRules(EntityInterface $entity, array $options = [], arr
      * Utility method for decorating any callable so that if it returns false, the correct
      * property in the entity is marked as invalid.
      *
-     * @param callable $rule The rule to decorate
+     * @param callable|\Cake\Datasource\RuleInvoker $rule The rule to decorate
```

Comment:
Adding `RuleInvoker` is redundant since it's an invokable class and thus sub type of `callable`.

Reply:
True, but here it still is helpful to clarify the concrete type expected as then you also have IDE support for the

    $rule->setOptions($options)->setName($name);
    
part.

Reply:
If people feel strongly about this not being explicit, I can also revert it.

Reply:
I dont mind it as long as no tool complains about redundant type.